src/openrct2/world/Park.cpp

```diff
@@ -657,7 +657,11 @@ uint8_t Park::CalculateGuestInitialHappiness(uint8_t percentage)
     // This sequence can be defined as PI*(9+n)/2 (the value is floored)
     for (uint8_t n = 1; n < 55; n++)
     {
-        if ((3.14159 * (9 + n)) / 2 >= percentage)
+        // Avoid floating point math by rescaling PI up.
+        constexpr int32_t SCALE = 100000;
+        constexpr int32_t PI_SCALED = static_cast<int32_t>(3.141590 * SCALE);
+        static_assert(PI_SCALED == 314159);
+        if (((PI_SCALED * (9 + n)) / SCALE) / 2 >= percentage)
```

Comment:
Why not just do `constexpr int32_t PI_SCALED = 314159`?

Reply:
To clarify that this is PI but I guess I could just comment it.

Reply:
Hmm, that would only make sense if you used the `M_PI` define.

Reply:
Changed it to the computed constant and added a comment to how it came to be.